core/src/main/scala/org/genivi/sota/core/daemon/DeltaListener.scala

```diff
@@ -0,0 +1,70 @@
+package org.genivi.sota.core.daemon
+
+import akka.Done
+import cats.data.Xor
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.core.UpdateService
+import org.genivi.sota.core.campaigns.CampaignLauncher
+import org.genivi.sota.core.db.{Campaigns, Packages}
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages.{DeltaGenerationFailed, GeneratedDelta}
+import slick.driver.MySQLDriver.api._
+import org.genivi.sota.core.data.Campaign
+import org.slf4j.LoggerFactory
+import org.genivi.sota.messaging.Commit.Commit
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeltaListener(deviceRegistry: DeviceRegistry, updateService: UpdateService,
+                    messageBus: MessageBusPublisher)(implicit db: Database)  {
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  /**
```

Comment:
In rvi-sota this is already a mess, but we shouldn't do this anywhere else and avoid doing it in sota if possible. `DBIO` instances should be contained to a class that handles database objects and should not leak ot of there. ie, they should stay inside repository classes and if needed communicate through `protected` methods.